WMF Framework/ReadingListsSyncOperation.swift

```diff
@@ -455,36 +455,37 @@ internal class ReadingListsSyncOperation: ReadingListsOperation {
             }
         }
         
-        let listNamesAndDescriptionsToCreate: [(name: String, description: String?)] = listsToCreate.flatMap {
-            guard !$0.isDefault, let name = $0.canonicalName else {
-                return nil;
-            }
-            return (name: name, description: $0.readingListDescription)
+
+        let listNamesAndDescriptionsToCreate: [(name: String, description: String?)] = listsToCreate.map {
+            assert($0.canonicalName != nil)
+            return (name: $0.canonicalName ?? "", description: $0.readingListDescription)
         }
         var start = 0
         var end = 0
         while end < listNamesAndDescriptionsToCreate.count {
             end = min(listNamesAndDescriptionsToCreate.count, start + WMFReadingListBatchSizePerRequestLimit)
             autoreleasepool {
                 taskGroup.enter()
-                self.apiController.createLists(listNamesAndDescriptionsToCreate, completion: { (readingListIDs, createError) in
+                let listsToCreateSubarray = Array(listsToCreate[start..<end])
+                let namesAndDescriptionsSubarray = Array(listNamesAndDescriptionsToCreate[start..<end])
+                self.apiController.createLists(namesAndDescriptionsSubarray, completion: { (readingListIDs, createError) in
                     defer {
                         taskGroup.leave()
                     }
                     guard let readingListIDs = readingListIDs else {
                         DDLogError("Error creating reading list: \(String(describing: createError))")
                         if let apiError = createError as? APIReadingListError {
-                            for list in listsToCreate {
+                            for list in listsToCreateSubarray {
                                 failedReadingLists.append((list, apiError))
                             }
                         }
                         return
                     }
                     for (index, readingList) in readingListIDs.enumerated() {
-                        guard (start + index) < listsToCreate.count else {
```

Comment:
Start isn't captured when this block is created, it's equivalent to an __block as a var, and it's value is past the end of the array when this callback is called.